src/app.js

```diff
@@ -91,48 +90,220 @@ app.routeAsync("/charon/*")
+ *
+ *   -trs, 25 Oct 2021
+ */
+app.use("/groups/:groupName", setGroupSource(req => req.params.groupName));
+
+app.routeAsync("/groups/:groupName")
+  /* sendGatsbyPage("groups/:groupName/index.html") should work, but it
+   * renders wrong for some reason that's not clear.
+   */
+  .getAsync(sendGatsbyEntrypoint)
+;
+
+// Avoid matching "narratives" as a dataset name.
+app.routeAsync("/groups/:groupName/narratives")
+  .getAsync((req, res) => res.redirect(`/groups/${esc(req.params.groupName)}`));
+
+app.routeAsync("/groups/:groupName/narratives/*")
+  .all(setNarrative(req => req.params[0]))
+  .getAsync(ifNarrativeExists, sendAuspiceEntrypoint)
+;
+
+app.routeAsync("/groups/:groupName/*")
+  .all(setDataset(req => req.params[0]))
+  .getAsync(ifDatasetExists, sendAuspiceEntrypoint)
+;
+
+
+/* Users
+ */
+app.routeAsync("/whoami")
+  .getAsync(users.getWhoami);
+
+/* For requests for text/html, the first /whoami implementation returned bare
+ * bones HTML dynamic on the logged in user.  This was later replaced by a
+ * redirect to /users/:name (for a logged in session) or /users (if no one
+ * logged in), which then rendered a Gatsby page.  However, the /users/:name
+ * form was purely aesthetic: :name was not used for routing at all.  Instead
+ * of keeping up this fiction, have /whoami render the Gatsby page directly,
+ * at least until we _actually_ implement user profile pages.
+ *   -trs, 4 Oct 2021
+ */
+app.route(["/users", "/users/:name"])
+  .get((req, res) => res.redirect("/whoami"));
+
+
+/* Auspice HTML pages and assets.
+ *
+ * Auspice hardcodes URL paths that start with /dist/… in its Webpack config,
+ * so we must use that prefix here too.
+ */
+app.route("/dist/*")
+  .all(expressStaticGzip(auspiceAssetPath(), {fallthrough: false, maxAge: '30d'}));
+
+
+/* Gatsby static HTML pages and other assets.
+ *
+ * Any URLs matching Gatsby's static files will be handled here, e.g.
+ * static-site/public/influenza/index.html will be served for /influenza.
+ *
+ * When a Gatsby dev server is in use, assets are proxied from the dev server
+ * instead of served straight from disk.
+ */
+if (app.locals.gatsbyDevUrl) {
+  // eslint-disable-next-line global-require, import/no-extraneous-dependencies
+  const {createProxyMiddleware} = require("http-proxy-middleware");
+
+  // WebSocket endpoint
+  app.get("/socket.io/", createProxyMiddleware(app.locals.gatsbyDevUrl, {ws: true}));
+
+  /* This is the end of the line in gatsbyDevUrl mode, as the proxy middleware
+   * doesn't fallthrough on 404 and even if it did the Gatsby dev server
+   * returns a 200 Ok for missing pages.
+   */
+  app.use(createProxyMiddleware(app.locals.gatsbyDevUrl));
+} else {
+  app.use(express.static(gatsbyAssetPath()));
 }
 
+
+/* Everything else gets 404ed.
+ */
+app.use((req, res, next) => next(new NotFound()));
+
+
+/* Error handler
+ */
+app.useAsync(async (err, req, res, next) => {
+  /* XXX TODO: Replace calls to Express' next() with calls to our own custom
```

Comment:
This would be great. At least with these changes we don't have traces for `NotFound` which (anecdotally) is the most common by far.

Reply:
Yeah, definitely want to come back to this. (but wanted to stop bundling up changes in order to get this out the door.)